uk_geo_utils/base_importer.py

```diff
@@ -375,3 +403,57 @@ def file_cleanup(self):
                 shutil.rmtree(self.tempdir)
             except OSError:
                 self.stdout.write("Failed to clean up temp files.")
+
+    def report_on_replicaton_status(self):
+        if not self.connection.settings_dict["HOST"]:
+            self.stdout.write(
+                "Command appears to have been run against a local database. Skipping replication checks."
+            )
+            return
+        dbname = self.cursor.db.connection.info.dbname
+        query_statement = f"""
+            SELECT slot_name, wal_status, active, conflicting 
+            FROM pg_replication_slots
+            WHERE database = '{dbname}';
+        """
+        self.cursor.execute(query_statement)
+        results = self.cursor.fetchall()
+        if not results:
+            self.stdout.write(
+                self.style.NOTICE(
+                    "No replication slots found. If you think there should be some either the --database flag was wrong, or something has broken.",
+                )
+            )
+            return
+
+        columns = [desc[0] for desc in self.cursor.description]
+        result_dicts = []
+        for row in results:
+            result_dicts.append(dict(zip(columns, row)))
+
+        self.stdout.write(self.style.SQL_TABLE("Replication status:"))
+        broken_slots = False
+        for result in result_dicts:
+            if result["active"] and not result["conflicting"]:
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"{result['slot_name']} is active and is not conflicting. 'wal_status' is {result['wal_status']}."
+                    )
+                )
+            else:
+                self.stdout.write(
+                    self.style.WARNING(
+                        f"{result['slot_name']} Doesn't seem healthy:"
+                        f"\n    'active': {result['active']}."
+                        f"\n    'wal_status': {result['wal_status']}."
+                        f"\n    'conflicting': {result['conflicting']}."
+                    )
+                )
+                broken_slots = True
+
+            if broken_slots:
+                self.stdout.write(
+                    self.style.NOTICE(
+                        "Some replication slots are broken. You can probably fix this by cycling out the instances."
+                    )
+                )
```

Comment:
This statement should be out-dented by one. At the moment it will run once on every iteration through the loop but we just want it to run once at the end.